File: scripts/sseqid_species_scraper.py
#!/usr/bin/python

from re import sub, search
import pandas as pd

family='MuDR'
outFile='/home/james/DDE_Pipeline/data/nr_pass_1/'+family+'_sseqid_species_nonunique.tsv'
inFile='/home/james/DDE_Pipeline/data/nr_pass_1/'+family+'/tbl/compiled_' +family+'.out'

with open(outFile, 'w') as out:
    with open(inFile, 'r') as blast_out:
        for line in blast_out:
            splitline = line.split('\t')
            sseqid = line.split()[1]
            stitle = splitline[15]
            if search('\\[', stitle):
              stitle=stitle.split('[')[1].split(' ')[0:2]
            else:
              stitle=stitle.split(' ')[0:2]
            stitle=' '.join(stitle)
            stitle=sub(']', '', stitle)
            out.write('\t'.join([sseqid, stitle]) + '\n')

File: scripts/sseqid_species_scraper.R
library(tidyverse)
library(stringr)

family <- "MuDR"

blast_out <- read_tsv(file = paste0("~/DDE_Pipeline/data/nr_pass_1/compiled_", family, ".out"), 
                      col_names = c("qseqid", "sseqid", "pident", "length", "mismatch",
                                    "gapopen", "qstart", "qend", "sstart", "send",
                                    "evalue", "bitscore", "species", "iteration",
                                    "qlen", "stitle", "Hit_seq"),
                      show_col_types = F) %>%
  dplyr::select(sseqid, stitle) %>%
  base::unique()
gc()

sseqid_species <- blast_out %>%
  mutate(species = sub(".*?\\[", "", stitle),
         species = word(species, start = 1L, end = 2L, sep = fixed(" ")),
         species = sub("\\]", "", species)) %>%
  dplyr::select(sseqid, species)

write_tsv(x = sseqid_species, file = paste0(family, "_sseqid_species.tsv"))